### PR TITLE
Rust implementation of strwidth() #334

### PR DESCRIPTION
**Pull request by zaowen**
Added a crate on this one that has a 'database' of widths.
lib.rs looks pretty diffrent from last time, let me know if what I've done is right.

### CODE REVIEW

File: rust/libnewsboat/src/utils.rs [hunk 1]
```diff
@@ -262,6 +264,16 @@ pub fn is_valid_attribute(attribute:  &str) -> bool {
     VALID_ATTRIBUTES.contains(&attribute)
 }
 
+pub fn strwidth( rs_str: &str) -> usize {
+    let control = rs_str.chars().fold( true, |acc, x| acc & !x.is_control());
+
+    if control {
+            return UnicodeWidthStr::width(rs_str);
```
Comment (Minoru) [inline rust/libnewsboat/src/utils.rs]: This is indented too far to the right.

File: rust/libnewsboat/Cargo.toml [hunk 2]
```diff
@@ -11,6 +11,7 @@ regex = "1.0"
 url = "1.7"
 dirs = "1.0.4"
 backtrace = "= 0.3"
+unicode-width = "0.1.5"
```
Comment (Minoru) [inline rust/libnewsboat/Cargo.toml]: Please commit an updated Cargo.lock, too. We added it to the repo just recently (#392); it ensures that we all use the same libraries when building Newsboat.

**Comment by Minoru** [comment]
Also, I'll be offline until 8th or 9th, so don't feel discouraged if I don't respond—I'll be back in a week and review everything you submit ;)

File: rust/libnewsboat/src/utils.rs (hunk 1 above)
Comment (zaowen) [reply rust/libnewsboat/src/utils.rs]: Thanks! good catch, vim has been indenting weird on rust lately.

File: rust/libnewsboat/Cargo.toml (hunk 2 above)
Comment (zaowen) [reply rust/libnewsboat/Cargo.toml]: Thank you. I didn't know that. I'll push it up.

**Comment by zaowen** [comment]
Take your time. I'm surprised you got to it as fast as you did. (Holidays and all that)

**Comment by Minoru** [comment]
Thank you for the work, @zaowen!

File: rust/libnewsboat/src/utils.rs [hunk 3]
```diff
@@ -262,6 +264,16 @@ pub fn is_valid_attribute(attribute:  &str) -> bool {
     VALID_ATTRIBUTES.contains(&attribute)
 }
 
+pub fn strwidth( rs_str: &str) -> usize {
+    let control = rs_str.chars().fold( true, |acc, x| acc & !x.is_control());
```
Comment (pickfire) [inline rust/libnewsboat/src/utils.rs]: @zaowen Did you mean
```rust
let control = rs_str.chars().any(|x| !x.is_control());
```

Comment (Minoru) [reply rust/libnewsboat/src/utils.rs]: @pickfire Thanks for taking the time to read through the code, but I don't think there is much use in reviewing a long-merged PR. Random improvements are always welcome, so if you spot stuff like this, just change it and submit a new PR.

**Pull request merged**